opsdroid/database/mongo/__init__.py

```diff
@@ -100,5 +100,14 @@ async def delete(self, key):
             key (str): the key is the database name.
 
         """
-        _LOGGER.debug("Deleting %s from MongoDB.", key)
-        return await self.database[self.collection].delete_one({"_id": key})
+        _LOGGER.debug("Deleting %s from MongoDB collection %s.", key, self.collection)
+
+        return await self.database[self.collection].delete_one({"key": key})
+
+    @contextmanager
+    def memory_in_collection(self, collection):
```

Comment:
If two different async tasks retrieve the db, and they each use a different collection, will they interfere with each other?
What if this returned a mutated copy of the db object instead of changing the setting like this?

Reply:
That actually sounds like a great idea to me.

Reply:
Hey @cognifloyd, so I made a modification to the `contextmanager` as requested. It works great, but needed to be changed to an `asynccontextmanager` as a result. I changed the docs to reflect usage, but it's basically as you suggested above. Thanks so much for the help. I'll see if I can make heads-or-tails of the tests, as they currently make no sense to me. Cheers.

Reply:
Nice that looks awesome. Maybe try commenting out the tests and slowly reimplements each one so that it makes sense?